### DISCUSSION TITLE
Question regarding independent timestamp variables support

### DISCUSSION THREAD

**Discussion by cliuj** [HadesArchitect/GrafanaCassandraDatasource#101] [Q&A] [answered]
Hi there,

I have a Cassandra schema where the `time column` is broken down into various columns identical to:
```
CREATE TABLE IF NOT EXISTS temperature (
    year int,
    month int,
    day int,
    minute,
    sensor_id int,
    temperature int,
    PRIMARY KEY (sensor_id)
);
```

Is it possible for the current `Query Editor` to support this format? Or would it be a good idea to add a feature to support this?

**Comment by HadesArchitect** [comment]
Hey @unflag , that should be possible with the #167 done?

**Reply by unflag** [reply]
No, datasource makes no assumptions about columns meaning, thus it cannot create timestamp by itself using set of int columns. C* has its own tools to do such things. It should be done by user - for example, converting that values to seconds and passing to `dateOf(maxTimeuuid())` functions. Or by using some user-defined function.

**Reply by HadesArchitect** [reply]
Yep, that's exactly what I mean. A user could do some computations or UDFs to get the first column combined of other columns, could he?

**Reply by unflag** [reply, marked as answer]
Certainly. If I were that user I would use UDF:) It is a quite powerful feature - basically, embedding Java/JavaScript code in cql queries.
